components/script/dom/document.rs

```diff
@@ -1196,6 +1199,53 @@ impl Document {
             .reflow(ReflowGoal::Full, ReflowReason::MouseEvent);
     }
 
+    #[allow(unsafe_code)]
+    pub fn handle_wheel_event(
+        &self,
+        js_runtime: *mut JSRuntime,
+        delta: WheelDelta,
+        client_point: Point2D<f32>,
+        node_Address: Option<UntrustedNodeAddress>,
+    ) {
+        let wheel_event_type_string = "wheel".to_owned();
+        debug!("{}: at {:?}", wheel_event_type_string, client_point);
+
+        let el = node_Address.and_then(|address| {
+            let node = unsafe { node::from_untrusted_node_address(js_runtime, address) };
+            node.inclusive_ancestors(ShadowIncluding::No)
+                .filter_map(DomRoot::downcast::<Element>)
+                .next()
+        });
+
+        let el = match el {
+            Some(el) => el,
+            None => return,
+        };
+
+        let node = el.upcast::<Node>();
```

Comment:
Not sure if you'd think it' easier to read if we use `if let` to extract the element?

```rust
let node = if let Some(el) = el {
  el.upcast::<Node>();
} else {
  return;
}
```

Reply:
Leaving this one the way it is